erica/request_processing/erica_input/v2/grundsteuer_input_gebaeude.py

```diff
@@ -0,0 +1,114 @@
+from typing import Optional
+
+from pydantic import root_validator
+
+from erica.request_processing.erica_input.v2.possibly_aliased_input import PossiblyAliasedInput
+
+
+class Ab1949(PossiblyAliasedInput):
+    is_ab1949: bool
+
+
+class Baujahr(PossiblyAliasedInput):
+    baujahr: Optional[str]
+
+
+class Kernsaniert(PossiblyAliasedInput):
+    is_kernsaniert: bool
+
+
+class Kernsanierungsjahr(PossiblyAliasedInput):
+    kernsanierungsjahr: Optional[str]
+
+
+class Abbruchverpflichtung(PossiblyAliasedInput):
+    has_abbruchverpflichtung: bool
+
+
+class Abbruchverpflichtungsjahr(PossiblyAliasedInput):
+    abbruchverpflichtungsjahr: Optional[str]
+
+
+class Wohnflaeche(PossiblyAliasedInput):
+    wohnflaeche: int
+
+
+class Wohnflaechen(PossiblyAliasedInput):
+    wohnflaeche1: int
+    wohnflaeche2: int
+
+
+class WeitereWohnraeume(PossiblyAliasedInput):
+    has_weitere_wohnraeume: bool
+
+
+class WeitereWohnraeumeDetails(PossiblyAliasedInput):
+    anzahl: int
+    flaeche: int
+
+
+class Garagen(PossiblyAliasedInput):
+    has_garagen: bool
+
+
+class GaragenAnzahl(PossiblyAliasedInput):
+    anzahl_garagen: int
+
+
+class Gebaeude(PossiblyAliasedInput):
+    ab1949: Ab1949
+    baujahr: Optional[Baujahr]
+    kernsaniert: Kernsaniert
+    kernsanierungsjahr: Optional[Kernsanierungsjahr]
+    abbruchverpflichtung: Abbruchverpflichtung
+    abbruchverpflichtungsjahr: Optional[Abbruchverpflichtungsjahr]
+    wohnflaeche: Optional[Wohnflaeche]
+    wohnflaechen: Optional[Wohnflaechen]
+    weitere_wohnraeume: WeitereWohnraeume
+    weitere_wohnraeume_details: Optional[WeitereWohnraeumeDetails]
+    garagen: Garagen
+    garagen_anzahl: Optional[GaragenAnzahl]
+
+    @root_validator
+    def baujahr_must_be_present_if_ab_1949(cls, values):
+        v = values.get("baujahr")
+        if values.get("ab1949").is_ab1949 is True and (not v or not v.baujahr):
```

Comment:
Instead of checking `not v.baujahr` here, we could just remove the Optional type on the field? So this way, gebaeude.baujahr can never be set without gebaeude.baujahr.baujahr being set?

Reply:
Brilliant idea! Those optionals are actually wrong in the first place...